demo/dbpedia/classification-example.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+#
+# Adapted from fastText/classification-example.sh
+
+myshuf() {
+  perl -MList::Util=shuffle -e 'print shuffle(<>);' "$@";
+}
+
+normalize_text() {
+  tr '[:upper:]' '[:lower:]' | sed -e 's/^\([0-9][0-9]*\),/\1 | /g' | \
+    sed -e "s/'/ ' /g" -e 's/"//g' -e 's/\./ \. /g' -e 's/<br \/>/ /g' \
+        -e 's/,/ , /g' -e 's/(/ ( /g' -e 's/)/ ) /g' -e 's/\!/ \! /g' \
+        -e 's/\?/ \? /g' -e 's/\;/ /g' -e 's/\:/ /g' | tr -s " " | myshuf
+}
+
+RESULTDIR=result
+DATADIR=data
+
+mkdir -p "${RESULTDIR}"
+mkdir -p "${DATADIR}"
+
+if [ ! -f "${DATADIR}/dbpedia.train" ]
+then
+  wget -c "https://github.com/le-scientifique/torchDatasets/raw/master/dbpedia_csv.tar.gz" -O "${DATADIR}/dbpedia_csv.tar.gz"
+  tar -xzvf "${DATADIR}/dbpedia_csv.tar.gz" -C "${DATADIR}"
+  cat "${DATADIR}/dbpedia_csv/train.csv" | normalize_text > "${DATADIR}/dbpedia.train"
+  cat "${DATADIR}/dbpedia_csv/test.csv" | normalize_text > "${DATADIR}/dbpedia.test"
+fi
+
+VW_EXEC=../../vowpalwabbit/vw
+
+${VW_EXEC} -d "${DATADIR}/dbpedia.train" \
+  --cache_file "${RESULTDIR}/dbpedia.cache" -f "${RESULTDIR}/dbpedia.bin" \
+  --oaa 14 --passes 5 --ngram 2 --loss_function hinge \
+  -k --threads
```

Comment:
No --bit_precision? It's likely to make a big difference there.

Reply:
I played around with the bit precision but it didn't change the result by much. the values i tried were 16 and 32. i tried 8 but that increases the error rate. is there some other value you think i should try?

one extra option available in fastText is the ability to specify a minimum count when building the vocabulary. that really doesn't fit with the way `vw` operates (i.e. using feature hashing) but i'm not sure if it helps fastText in terms of generalization or not.

Reply:
Try anything above 18 (the default) e.g. `-b 20`, you may go as high as 31, I think but at really high values things may get significantly slower when storing the model.

Reply:
Updated based on your comments (I also played around with regularization and skips) PTAL.